cmd/gcp-controller-manager/kubelet_readonly_csr_approver_test.go

```diff
@@ -116,18 +162,26 @@ func TestKubeletReadonlyApprover(t *testing.T) {
 
 	for _, testCase := range testCases {
 		t.Run(fmt.Sprint(testCase.name), func(t *testing.T) {
-			request := generateTestRequest(t, testCase.sarAllowed, testCase.podAnnotationAllowed, testCase.usageAllowed, testCase.autopilotEnabled)
+			request := generateTestRequest(t, testCase.sarAllowed, testCase.podAnnotationAllowed, testCase.usageAllowed, testCase.autopilotEnabled,
+				testCase.sarErrored, testCase.podAnnotationErrored)
+			request.csr.Name = testCase.name
 			request.csr.Spec.SignerName = kubeletReadonlyCSRSignerName
 			approver := newKubeletReadonlyCSRApprover(request.controllerContext)
 			*autopilotEnabled = testCase.autopilotEnabled
 			err := approver.handle(request.context, request.csr)
 			if err != nil {
-				t.Fatalf("error when handle approver, error: %v", err)
-			}
-			if diff := cmp.Diff(request.csr.Status.Conditions[0], testCase.expectCondition, cmp.Comparer(func(x, y capi.CertificateSigningRequestCondition) bool {
-				return x.Type == y.Type && x.Reason == y.Reason && x.Message == y.Message && x.Status == y.Status
-			})); diff != "" {
-				t.Fatalf("condition don't match, diff -want +got\n%s", diff)
+				if testCase.expectError == nil {
+					t.Fatalf("error when handle approver, error: %v", err)
+				}
+				if err.Error() != testCase.expectError.Error() {
+					t.Fatalf("error not match, expect: %v, got: %v", testCase.expectError, err)
+				}
+			} else {
```

Comment:
I don't think this catches the case where err is nil but testCase.exepectError is not nil